src/internal/mod.rs

```diff
@@ -1027,6 +1053,43 @@ pub(crate) mod test {
             unimplemented!()
         }
     }
+    // impl CryptoOps for MockKeyGenOps {
```

Comment:
What's going on with this test?

Reply:
It's currently commented out as I don't know how to mock the CryptoOps needed for `transform::gen_group_keys`. I will either need help writing it, or suggestions on an alternative.

Reply:
It's now implemented. I left a comment on why I was unable to test that it retries if the private key is zero, but that should be covered in the other unit test that calls the same `augment` function.